pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -660,14 +660,20 @@ def _get_profiles(profiles):
         pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
         pulp2_ids = pulp2_id_obj_map.keys()
         pulp2_content_batch = pulp2_models.ModulemdDefaults.objects.filter(id__in=pulp2_ids)
-        pulp2defaults_to_save = [
-            cls(module=defaults.name,
-                stream=defaults.stream,
-                profiles=_get_profiles(defaults.profiles),
-                digest=defaults.checksum,
-                repo_id=defaults.repo_id,
-                pulp2content=pulp2_id_obj_map[defaults.id])
-            for defaults in pulp2_content_batch]
+        pulp2defaults_to_save = []
+
+        for defaults in pulp2_content_batch:
+            kwargs = {
+                'module': defaults.name,
+                'digest': defaults.checksum,
+                'repo_id': defaults.repo_id,
+                'pulp2content': pulp2_id_obj_map[defaults.id]
+            }
+            if defaults.stream:
+                kwargs['stream'] = defaults.stream
+            if defaults.profiles:
```

Comment:
_get_profiles(defaults.profiles) if defaults.profiles else []

Reply:
```
n [8]: Pulp2ModulemdDefaults(module='test2',pulp2content_id=p.pk).save()
---------------------------------------------------------------------------
NotNullViolation                          Traceback (most recent call last)
/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/backends/utils.py in _execute(self, sql, params, *ignored_wrapper_args)
     83             else:
---> 84                 return self.cursor.execute(sql, params)
     85 

NotNullViolation: null value in column "profiles" violates not-null constraint
DETAIL:  Failing row contains (80f2125a-b790-4de7-8ab1-98ef7362a636, 2021-01-14 17:23:34.868795+00, 2021-01-14 17:23:34.868838+00, test2, , null, , , 0329f8c7-e75b-4eb6-b39d-542ff6be7bc8).
```

Reply:
I added it to the function instead.

Reply:
No what i meant is that in case profiles will be None  in defaults.profiles the kwargs will not have 'profiles' key at all so when saving pulp2modulemddefaults it will blow up with
`NotNullViolation: null value in column "profiles" violates not-null constraint`

Reply:
What I mean is that I did this:

https://github.com/pulp/pulp-2to3-migration/pull/293/files#diff-669c91fe94bfb50613be9955c7cce760e8a0ee962b99c60544321bfd82783a4fR658

It's exactly equivalent to your suggestion just in a different place

Reply:
sorry i needed to be more explicit. I suggest removing ` if defaults.profiles:`
and have this in kwargs
```
            kwargs = {
                digest=defaults.checksum,
                'module': defaults.name,
                repo_id=defaults.repo_id,
                'digest': defaults.checksum,
                pulp2content=pulp2_id_obj_map[defaults.id])
                'repo_id': defaults.repo_id,
                'pulp2content': pulp2_id_obj_map[defaults.id]
               'profiles': _get_profiles(defaults.profiles) if defaults.profiles else []
}

```

Reply:
I think ```profiles``` is actually supposed to be a dict due to how it's stored in Pulp 2.  With the default value fixed properly, it seems like it works now.

If we're going to do a migration anyway I'm going to look for any other places where we might have made this mistake.